framework/caching/Cache.php

```diff
@@ -202,14 +207,20 @@ public function multiGet($keys)
      * @param mixed $key a key identifying the value to be cached. This can be a simple string or
      * a complex data structure consisting of factors representing the key.
      * @param mixed $value the value to be cached
-     * @param int $duration the number of seconds in which the cached value will expire. 0 means never expire.
+     * @param int $duration default duration in seconds before the cache will expire. Default value is 0, meaning infinity.
      * @param Dependency $dependency dependency of the cached item. If the dependency changes,
      * the corresponding value in the cache will be invalidated when it is fetched via [[get()]].
      * This parameter is ignored if [[serializer]] is false.
      * @return bool whether the value is successfully stored into cache
      */
     public function set($key, $value, $duration = 0, $dependency = null)
     {
+        $args = func_get_args();
+        if (!isset($args[2])) {
```

Comment:
this looks really strange. I'd prefer making `$duration = null` the default and if it is `null` use `$this->ttl`.